src/lib.rs

```diff
@@ -32,13 +32,21 @@ lalrpop_mod!(#[allow(clippy::all)] pub parser);
 /// Compile a complete system policy
 /// The list of input files list should contain filenames of files containing policy to be
 /// compiled.
+/// The list of system names are the names of the systems to build.
+/// The dummy system is used to build if no system names are given.
 /// Returns a Result containing either a string of CIL policy which is the compiled result or a
 /// list of errors.
-/// In order to convert the compiled CIL policy into a usable policy, you must use secilc
-pub fn compile_system_policy(input_files: Vec<&str>) -> Result<String, error::CascadeErrors> {
+/// In order to convert the compiled CIL policy into a usable policy, you must use secilc.
+pub fn compile_system_policies(
+    input_files: Vec<&str>,
+    mut system_names: Vec<String>,
+    dummy_system: Option<ValidatedSystem>,
```

Comment:
I don't think we want to expose ValidatedSystem outside the library.  It looks like casc calls this as None anyways, and the current use case is just for compile_all_system_policies(), right?

The issue here is that compile_system_policies() is `pub`, so it gets exposed to users, and should therefore have a really clean interface.

Off the top of my head, I think the cleanest thing might be to have compile_system_policies() just take input files and system names, and then directly call into a new function compile_system_policies_internal(), which does what compile_system_policies does now.  compile_all_system_policies() can then also call into the _internal() version.

If system_names is empty, we can return an error.

I think "default" is probably a better name than "dummy".

Reply:
In response to "If system_names is empty, we can return an error," I want to note that with the current implementation, since -s has a default value, there is no error for when the user does not provide system names.
I assume this will be changed, so that -s will no longer have a default value, after we figure out how we want to specify building all systems. However, in that case, clap has its own error for not supplying a value.